### PR TITLE
Dev refactor com unittests

### LINKED ISSUE #281: Dev refactor COM unit tests
COM module is using GTest only atm. Shall be using GMock.

### PR DESCRIPTION
**Pull request by c0nwild**
Closes #281 

Unittests for the com component were disabled due to changes in the SPI interface. Furthermore, com interface unit tests did not use google Mock. Refactoring necessary. 

During refactoring, some minor changes to the com component were implemented as well (test driven development) and are now part of this CR:
- COM buffer returns an error in case the data frame is too large.

Acceptance criteria:

- Unit tests cover 100% and all pass
- Unit tests are complete (no logical fallacy is untested and all equivalency classes were tested)
- COM class diagram from wiki is not in conflict with implementation
- All automatic checks pass
- Google Styleguide was followed
- Additional guidelines from the wiki were followed
- No logical errors in code
- All error return values are always checked

### CODE REVIEW

File: tests/com/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +11,7 @@ add_testpackage(TEST_NAME
 
 add_testpackage(TEST_NAME 
                     com_message_buffer 
-                SOURCES 
+                    SOURCES
```
Comment (c0nwild) [inline tests/com/CMakeLists.txt]: Correct indentation.

File: tests/com/com_interface_tests.cpp [hunk 2]
```diff
@@ -22,30 +26,32 @@ class ConcreteComInterface final : public com::ComInterface {
...
+  auto unit_under_test = std::make_unique<ConcreteComInterface>(std::move(com_buffer_));
   ASSERT_EQ(data_compare, unit_under_test->GetDataPacket());
+  ASSERT_THAT(unit_under_test->GetDataPacket(), data_compare);
 }
 
 TEST_F(ComInterfaceTests, put_data_packet) {
   types::com_msg_frame data;
   auto com_buffer = std::make_unique<com::ComMessageBuffer>();
-  auto unit_under_test = std::make_unique<ConcreteComInterface>(std::move(com_buffer));
+  auto unit_under_test = std::make_unique<ConcreteComInterface>(std::move(com_buffer_));
```
Comment (c0nwild) [inline tests/com/com_interface_tests.cpp:53]: Why not the buffer above?

**Review by c0nwild — COMMENTED** [review]
Changes necessary

**Review by Kupfa — APPROVED** [review]
LGTM

**Pull request merged**